shuffle/cli/src/transactions.rs

```diff
@@ -0,0 +1,227 @@
+                old_seq_num = write_out_txns(
+                    json_with_txns,
+                    *last_txn_seq_num,
+                    old_seq_num,
+                    &mut io::stdout(),
+                )?;
+            }
+        }
+    }
+    Ok(())
+}
+
+pub fn write_out_txns<W: Write>(
+    json: Value,
+    last_txn_seq_num: u64,
+    old_seq_num: u64,
+    mut stdout: W,
+) -> Result<u64> {
+    // Only want to write the first transaction (seq num: 0) or
+    // transactions where the sequence number changes
+    if last_txn_seq_num == 0 || last_txn_seq_num > old_seq_num {
+        let all_transactions = json
+            .as_array()
+            .ok_or_else(|| anyhow::anyhow!("Failed to get transactions"))?;
+
+        // setting to 1 to exclude the index of the txn we've already printed out
+        let mut start_ind = 1;
+        if last_txn_seq_num == 0 {
+            // setting to 0 to include the first txn.
+            // Only needed for the very first txn
+            start_ind = 0;
+        }
+        for txn_index in start_ind..all_transactions.len() {
+            write_into(&mut stdout, &json[txn_index])?;
+            println!();
+            println!();
+        }
+    }
+    Ok(last_txn_seq_num)
+}
+
+pub async fn get_account_txns_response(
+    client: &Client,
+    path: &Url,
+    start: u64,
+    limit: u64,
+) -> Result<Response> {
+    Ok(client
+        .get(path.as_str())
+        .query(&[("start", start.to_string().as_str())])
+        .query(&[("limit", limit.to_string().as_str())])
+        .send()
+        .await?)
+}
+
+pub async fn get_sequence_number(
+    client: &Client,
+    network: &Url,
+    address: AccountAddress,
+) -> Result<u64> {
+    let path =
+        network.join(format!("accounts/{}/resources/", address.to_hex_literal()).as_str())?;
+    let resp = client.get(path.as_str()).send().await?;
+    let json: Vec<Value> = serde_json::from_str(resp.text().await?.as_str())?;
+    parse_json_for_seq_num(json)
+}
+
+pub fn parse_json_for_seq_num(json_objects: Vec<Value>) -> Result<u64> {
+    let mut seq_number_string = "";
+    for object in &json_objects {
+        if object["type"]["name"] == "DiemAccount" {
+            seq_number_string = object["value"]["sequence_number"]
+                .as_str()
+                .ok_or_else(|| anyhow!("Invalid sequence number string"))?;
+            break;
+        };
+    }
+    let seq_number: u64 = seq_number_string.parse()?;
+    Ok(seq_number)
+}
+
+fn write_into<W>(writer: &mut W, json: &serde_json::Value) -> Result<()>
+where
+    W: Write,
+{
+    write!(writer, "{}", json)?;
+    Ok(())
+}
+
+#[cfg(test)]
+mod test {
+    use crate::transactions::{parse_json_for_seq_num, write_into, write_out_txns};
+    use serde_json::{json, map::Map, Value};
+
+    #[test]
+    fn test_parse_json_for_seq_num() {
+        let mut name_map: Map<String, Value> = Map::new();
+        name_map.insert("name".parse().unwrap(), Value::from("DiemAccount"));
```

Comment:
Rather than manually create a map, have a json string similar to actual input that you parse to improve readability. @xli has a great example here: 

https://github.com/dimroc/diem/blob/8fab7b7f2e131bdcfad475b0509c636e5e1883bf/api/src/tests/accounts_test.rs#L78-L102

Reply:
will do!